src/test/java/org/jabref/logic/openoffice/oocsltext/CSLFormatUtilsTest.java

```diff
@@ -0,0 +1,724 @@
+                Arguments.of(
+                        "Text with <span style=\"text-decoration:underline;\">underline</span>",
+                        "Text with <u>underline</u>"
+                ),
+
+                // Replacing span tags with inline styles for small-caps
+                Arguments.of(
+                        "Text with <span style=\"font-variant:small-caps;\">small caps</span>",
+                        "Text with <smallcaps>small caps</smallcaps>"
+                ),
+
+                // Test case for cleaning up remaining span tags
+                Arguments.of(
+                        "Text with <span>unnecessary span</span>",
+                        "Text with unnecessary span"
+                ),
+
+                // Test case combining multiple transformations
+                Arguments.of(
+                        "<div class=\"csl-left-margin\">[1]</div><div class=\"csl-right-inline\"><span style=\"font-weight:bold;\">Author</span>, &quot;Title,&quot; <span style=\"font-style:italic;\">Journal</span>, vol. 1, no. 1, pp. 1-10, 2023.</div>",
+                        "[1] <b>Author</b>, \"Title,\" <i>Journal</i>, vol. 1, no. 1, pp. 1-10, 2023."
+                ),
+
+                // Comprehensive test
+                Arguments.of(
+                        "<div class=\"csl-left-margin\">[1]</div><div class=\"csl-right-inline\">" +
+                                "<span style=\"font-weight:bold;\">Smith&nbsp;&amp;&nbsp;Jones</span>, " +
+                                "&quot;<span style=\"font-style:italic;\">Analysis of &lt;code&gt; in HTML</span>,&quot; " +
+                                "<span style=\"font-variant:small-caps;\">Journal of Web&nbsp;Development</span>, " +
+                                "vol. 1, no. 1, pp. 1-10, 2023. " +
+                                "<a href=\"https://doi.org/10.1000/example\">https://doi.org/10.1000/example</a></div>",
+
+                        "[1] <b>Smith & Jones</b>, " +
+                                "\"<i>Analysis of <code> in HTML</i>,\" " +
+                                "<smallcaps>Journal of Web Development</smallcaps>, " +
+                                "vol. 1, no. 1, pp. 1-10, 2023. " +
+                                "https://doi.org/10.1000/example"
+                )
+        );
+    }
+
+    /**
+     * Test to check correct transformation of raw CSL bibliography generated by citeproc-java methods into OO-ready text.
+     * <p>
+     * <b>Precondition:</b> This test assumes that {@link CitationStyleGenerator#generateCitation(List, String, CitationStyleOutputFormat, BibDatabaseContext, BibEntryTypesManager) generateCitation} works as expected.
+     * </p>
+     *
+     * @param style the CSL style to test transformation on.
+     */
+    @ParameterizedTest
+    @MethodSource("rawBibliographyProvider")
+    void ooHTMLTransformFromRawBibliography(CitationStyle style, String expected) {
```

Comment:
To be in line with `assertEquals`, the `expected` parameter should be passed first.

Reply:
So should this be flipped in every method (along with the provider arguments)?

Reply:
Okay, changed uniformly on all methods and providers.